packages/vscode-dependencies-validation/test/diagnostics.spec.ts

```diff
@@ -0,0 +1,84 @@
+import proxyquire = require("proxyquire");
+import { diagnosticCollection } from "./vscodeMocks";
+import { createSandbox, SinonMock, SinonSandbox } from "sinon";
+import { refreshDiagnostics } from "../src/diagnostics";
+import { range } from "lodash";
+import { NPM_DEPENDENCY_ISSUES_CODE } from "../src/constants";
+
+describe("diagnostics unit test", () => {
+  const packageJsonPath = "/root/folder/package.json";
+  const Uri = {
```

Comment:
on first look I think the "fuller" mocks with actual basic implementation are cleaner and better as we don't have to assume
the code does not actually need implementation details.